eclipse-collections-api/src/main/java/org/eclipse/collections/api/factory/map/ImmutableMapFactory.java

```diff
@@ -59,12 +60,37 @@
 
     <K, V> ImmutableMap<K, V> with(K key1, V value1, K key2, V value2, K key3, V value3, K key4, V value4);
 
+    /**
+     * @deprecated use {@link #withMap(Map)} instead (inlineable).
+     */
+    @Deprecated
     <K, V> ImmutableMap<K, V> ofMap(Map<K, V> map);
 
     /**
      * Same as {@link #withAll(Map)}.
+     * @deprecated since 10.3.0, use {@link #withMap(Map)} instead.
      */
+    @Deprecated
     <K, V> ImmutableMap<K, V> ofAll(Map<K, V> map);
 
+    /**
+     * @deprecated since 10.3.0, use {@link #withMap(Map)} instead.
+     */
+    @Deprecated
     <K, V> ImmutableMap<K, V> withAll(Map<K, V> map);
+
+    /**
+     * @since 10.3.0
+     */
+    <K, V> ImmutableMap<K, V> withMap(Map<K, V> map);
+
+    /**
+     * @since 10.3.0
+     */
+    <K, V> ImmutableMap<K, V> ofMapIterable(MapIterable<K, V> mapIterable);
+
+    /**
+     * @since 10.3.0
+     */
+    <K, V> ImmutableMap<K, V> withMapIterable(MapIterable<K, V> mapIterable);
```

Comment:
ImmutableMapFactoryImpl has a single deprecated method.

```java
    /**
     * @deprecated use {@link #ofAll(Map)} instead (inlineable)
     */
    @Override
    @Deprecated
    public <K, V> ImmutableMap<K, V> ofMap(Map<K, V> map)
    {
        return this.ofAll(map);
    }
```

This does not bring the interface in line with the implementation. The one method that ought to be deprecated doesn't match the correct suggestion.